crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -263,7 +269,7 @@ impl ObjectType {
             ObjectType::NotAnObject => 0,
             ObjectType::Signed(c) => *c,
             ObjectType::Unsigned(c) => *c,
-            ObjectType::Pointer(_) => unreachable!(),
+            ObjectType::Pointer(_) => 0,
```

Comment:
When was this `unreachable` triggering before? What effects will returning 0 for pointers have here?

Reply:
The function returns the number of bits that the type is using. A pointer does not consume any bit because it is a reference to a memory location, which will be ultimately de-referenced. 
However, functions from the standard lib (i.e using Plonk custom gates) are not resolved by the compiler because they are implemented inside barretenberg and so they may use pointer as input or output types, which will reach the overflow strategy step. This step needs to know the bit size of every variable.